homeassistant/components/spokestack/config_flow.py

```diff
@@ -0,0 +1,58 @@
+"""ConfigFlow for Spokestack."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+
+from .const import CONF_IDENTITY, CONF_SECRET_KEY, DEFAULT_LANG, DOMAIN
+from .tts import CONF_LANG, get_engine
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_IDENTITY, default="Identity"): str,
+        vol.Required(CONF_SECRET_KEY, default="Secret Key"): str,
+        vol.Optional(CONF_LANG, default=DEFAULT_LANG): str,
+    }
+)
+
+
+async def validate_user_input(hass, user_input):
+    """Validate the user's API keys."""
+    # Setup a client to ping the API.
+    tts = get_engine(hass, user_input)
+    # Verify synthesis does not result in error.
+    await hass.async_add_executor_job(tts.get_tts_audio, "ping")
```

Comment:
Is there no other way of checking if the key works ?

Reply:
Unfortunately, trying a synthesis request is the only way to validate the token.